osu.Game/Screens/Backgrounds/BackgroundScreenDefault.cs

```diff
@@ -22,23 +26,36 @@ public class BackgroundScreenDefault : BackgroundScreen
 
         private string backgroundName => $@"Menu/menu-background-{currentDisplay % background_count + 1}";
 
-        private Bindable<User> user;
-        private Bindable<Skin> skin;
+        public readonly Bindable<MainMenuBackgroundMode> BackgroundMode = new Bindable<MainMenuBackgroundMode>();
+        public Bindable<User> User = new Bindable<User>();
+        public Bindable<Skin> Skin = new Bindable<Skin>();
+
+        private readonly Bindable<WorkingBeatmap> beatmap = new Bindable<WorkingBeatmap>();
+        private WorkingBeatmap currentBeatmap;
 
         [BackgroundDependencyLoader]
-        private void load(IAPIProvider api, SkinManager skinManager)
+        private void load(IAPIProvider api, SkinManager skinManager, OsuConfigManager config, Bindable<WorkingBeatmap> workingBeatmap)
         {
-            user = api.LocalUser.GetBoundCopy();
-            skin = skinManager.CurrentSkin.GetBoundCopy();
-
-            user.ValueChanged += _ => Next();
-            skin.ValueChanged += _ => Next();
+            Skin = skinManager.CurrentSkin.GetBoundCopy();
+            beatmap.BindTo(workingBeatmap);
+            User = api.LocalUser.GetBoundCopy();
+            config.BindWith(OsuSetting.MenuBackgroundMode, BackgroundMode);
 
             currentDisplay = RNG.Next(0, background_count);
 
             display(createBackground());
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            User.BindValueChanged(onUserChanged, true);
```

Comment:
i'm 99.9% sure this is going to trigger multiple background reloads on first run.